spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,387 @@
-        get :index, category_id: model.category
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).include?(mod_same_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).size).to eq(1)
-      end
-    end
-    context 'without @category set' do
-      it 'should populate an array of all active equipment models' do
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).size).to eq(2)
-      end
-    end
+  def mock_eq_collection(models: nil, count: 1)
+    models ||= Array.new(count) { mock_eq_model }
+    allow(EquipmentModel).to receive(:all).and_return(models)
+    allow(EquipmentModel).to receive(:where).and_return([])
+    models
   end
-  context 'with show deleted' do
-    let!(:mod_other_cat_active) { FactoryGirl.create(:equipment_model) }
-    let!(:mod_other_cat_inactive) do
-      FactoryGirl.create(:equipment_model,
-                         deleted_at: Time.zone.today)
-    end
-    context 'with @category set' do
-      it 'should populate an array of category-type equipment models' do
-        mod_same_cat_inactive =
-          FactoryGirl.create(:equipment_model, category: model.category,
-                                               deleted_at: Time.zone.today)
-        get :index, category_id: model.category, show_deleted: true
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).include?(mod_same_cat_inactive))
-          .to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).size).to eq(2)
-      end
-    end
-    context 'without @category set' do
-      it 'should populate an array of all equipment models' do
-        get :index, show_deleted: true
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .to be_truthy
-        expect(assigns(:equipment_models).size).to eq(3)
-      end
-    end
+
+  def mock_category(**attrs)
+    cat = instance_spy('Category', **attrs)
+    allow(Category).to receive(:find).and_return(cat)
```

Comment:
this is bad, should probably generate unique category ids like equipment model ids to fix